openmrs-android-sdk/src/main/java/com/openmrs/android_sdk/library/api/repository/AppointmentRepository.kt

```diff
@@ -0,0 +1,202 @@
+ * Version 1.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ * http://license.openmrs.org
+ *
+ * Software distributed under the License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
+ * License for the specific language governing rights and limitations
+ * under the License.
+ *
+ * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
+ */
+package com.openmrs.android_sdk.library.api.repository
+
+import com.openmrs.android_sdk.R
+import com.openmrs.android_sdk.library.databases.AppDatabase
+import com.openmrs.android_sdk.library.databases.AppDatabaseHelper
+import com.openmrs.android_sdk.library.databases.AppDatabaseHelper.createObservableIO
+import com.openmrs.android_sdk.library.models.*
+import rx.Observable
+import java.io.IOException
+import java.util.concurrent.Callable
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class AppointmentRepository @Inject constructor() : BaseRepository(){
+
+    val representation =  context.resources.getString(R.string.appointment_resource_representation)
+    var appointmentRoomDAO = AppDatabase.getDatabase(context).appointmentRoomDAO()
+
+
+    /**
+     * Creates an appointment block
+     *
+     * @param startDate the start date of the block
+     * @param endDate the end date of the block
+     * @param locationUUID the location name of the block
+     * @param types the list of appointment types
+     *
+     * @return the AppointmentBlock object
+     */
+    fun createAppointmentBlock(
+        startDate: String,
+        endDate: String,
+        locationUUID: String,
+        types: List<AppointmentType>
+    ): Observable<AppointmentBlock> {
+        return createObservableIO<AppointmentBlock>(Callable {
+            val call = restApi.createAppointmentBlock(startDate, endDate, locationUUID, types)
+            val response = call.execute()
+
+            if (response.isSuccessful && response != null) {
+                return@Callable response.body()!!
+            } else {
+                logger.e("Error creating an appointment block: " + response.message())
+                throw Exception(response.message())
+            }
+        }
+        )
+    }
+
+    /**
+     * Creates a time slot
+     *
+     * @param timeSlot the TimeSlot object
+     *
+     * @return the TimeSlot object
+     */
+    fun createTimeSlot(timeSlot: TimeSlot): Observable<TimeSlot>{
+        return createObservableIO<TimeSlot>(Callable {
+            val call = restApi.createTimeslot(timeSlot)
+            val response = call.execute()
+
+            if (response.isSuccessful && response != null) {
+                return@Callable response.body()!!
+            } else {
+                logger.e("Error creating Time Slot: " + response.message())
+                throw Exception(response.message())
+            }
+        }
+        )
```

Comment:
```suggestion
        }
        )
```
```suggestion
        })
```
Add this change in all remaining functions